pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -13,6 +13,18 @@
 from pandas.core.groupby.grouper import Grouping
 
 
+def test_groupby_agg_no_extra_calls():
+    # GH#31760
+    df = pd.DataFrame({"key": ["a", "b", "c", "c"], "value": [1, 2, 3, 4]})
+    gb = df.groupby("key")["value"]
+
+    def dummy_func(x):
```

Comment:
I think a better approach here would be to Mock a function and examine the call_count after application. There's an example using `create_autospec` in the Python docs

https://docs.python.org/3/library/unittest.mock.html

Reply:
we already use this pattern elsewhere